python/plugins/processing/algs/gdal/gdalcalc.py

```diff
@@ -57,41 +67,98 @@ def __init__(self):
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_B,
+                self.tr('Number of raster band for B'),
+                parentLayerParameterName=self.INPUT_B, 
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_C,
+                self.tr('Input layer C'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(self.BAND_C,
+                self.tr('Number of raster band for C'),
+                parentLayerParameterName=self.INPUT_C,
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_D,
+                self.tr('Input layer D'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_D,
+                self.tr('Number of raster band for D'),
+                parentLayerParameterName=self.INPUT_D,
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_E,
+                self.tr('Input layer E'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_E,
+                self.tr('Number of raster band for E'),
+                parentLayerParameterName=self.INPUT_E,
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_F, 
+                self.tr('Input layer F'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_F,
+                self.tr('Number of raster band for F'),
+                parentLayerParameterName=self.INPUT_F,
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterString(self.FORMULA,
+                self.tr('Calculation in gdalnumeric syntax using +-/* or any numpy array functions (i.e. logical_and())'), 
+                'A*2', 
+                optional=False))
+        self.addParameter(
+            QgsProcessingParameterString(self.NO_DATA,
+                self.tr('Set output nodata value'),
+                '', 
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterEnum(
+                self.RTYPE,
+                self.tr('Output raster type'),
+                options=self.TYPE,
+                defaultValue=5))
         #self.addParameter(ParameterBoolean(
         #    self.DEBUG, self.tr('Print debugging information'), False))
-        self.addParameter(ParameterString(self.EXTRA,
-                                          self.tr('Additional creation parameters'), '', optional=True))
-        self.addOutput(OutputRaster(self.OUTPUT, self.tr('Calculated')))
+        self.addParameter(
```

Comment:
Set the metadata for this param to get the nice widget - e.g. copy https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/gdal/merge.py#L93